README
NAME

    Dist::Zilla::TravisCI - Travis CI plugins for your DZIL-based
    distribution

SYNOPSIS

        [@Prereqs]
        [TravisYML]
     
        [Git::CommitBuild]
        release_branch  = build/%b
        release_message = Release build of v%v (on %b)
     
        [@Git]
        allow_dirty = dist.ini
        allow_dirty = README
        allow_dirty = .travis.yml
        push_to = origin
        push_to = origin build/master:build/master

DESCRIPTION

 What is Travis CI?

    Travis CI <https://travis-ci.org/> is pure awesome! Travis is a free
    and open source CI platform that hooks directly into GitHub
    <https://github.com/>. It's extremely easy to configure, tests on every
    major 5.10+ Perl version, and will notify you of test results, either
    through email or IRC. (Yes, a bot will jump into an IRC room, tell you
    the results, and bounce. How cool is that?!)

    If you're not familiar with CI platforms, this is your chance to get
    your feet wet. CI stands for Continuous Integration
    <http://en.wikipedia.org/wiki/Continuous_integration>. This kind of
    testing is designed around testing your code every time you commit/push
    a change. In this case, this is a hook into Git via GitHub to make sure
    that any change you make is going to pass tests.

 Perl is already a "test-heavy" community. What does this add?

    This is true. Travis CI has a much more profound use with languages
    without a deep-seated testing platform like Perl's t/* directory (and
    its hooks with CPAN).

    However, adding Travis CI to your test platform has its benefits:

      * Testing in a plain vanilla environment, which your work environment
      is most definitely NOT.

      * Testing with Perl 5.10, 5.12, 5.14, 5.16, and beyond, all at the
      same time.

      * Testing your DZIL environment, which not even CPAN can do.

      * Use Windows or some other non-Linux OS for your work environment?
      Great. Travis runs on Ubuntu, so you can confirm that it works on
      Linux. (Eventually, Travis will support other OSs, like Windows, to
      make this even sweeter!)

      * Minimum Version Dependency Testing

      * Public announcements of test results. If somebody breaks something
      in master, make sure EVERYBODY knows it!

      * If nothing else, you get many more tests before it even touchs
      CPAN.

 What about the automated CPAN testers?

    They're great! But they are also a bit too late in the process. Sure,
    you can fix what you find after you've found something, but you want
    some extra insurance that your production release is actually
    production ready.

    Development releases on CPAN also work, but you have no idea what is on
    those CPAN tester platforms. Does it cover the full gamut of Perl
    versions, or are they just grabbing the latest? Are they all using the
    latest module dependencies (which may not be want you want)?

    You have more control over your Travis CI platform, so you can set up
    the environment the way you want it. That's not to say that the massive
    CPAN tester army isn't a great boon for Perl, but in the fight against
    bugs, you want every weapon available.

 This sounds complicated...

    Actually, it's the polar opposite of complicated. This is literally a
    full guide on how to make it work:

      1. Sign into Travis CI <http://travis-ci.org/users/auth/github> with
      your GitHub account.

      2. Flip a bit on your distro through the profile page
      <https://travis-ci.org/profile>.

      3. Put the TravisYML plugin in your dist.ini.

    There's some extra configuration you can do: build branches, MVDT, etc.
    But, that's the basic setup. Easy!

DONATIONS

    Travis CI is a completely free service, and they put in a lot of hard
    work to make it what it is.

    Please donate! <https://love.travis-ci.org/>

TODO

      * MVDT - Yes, I actually need to finish MVDT on my own distros.

AVAILABILITY

    The project homepage is
    https://github.com/SineSwiper/Dist-Zilla-TravisCI.

    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit http://www.perl.com/CPAN/ to find a
    CPAN site near you, or see
    https://metacpan.org/module/Dist::Zilla::TravisCI/.

SUPPORT

 Internet Relay Chat

    You can get live help by using IRC ( Internet Relay Chat ). If you
    don't know what IRC is, please read this excellent guide:
    http://en.wikipedia.org/wiki/Internet_Relay_Chat. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You
    can join those networks/channels and get help:

      * irc.perl.org

      You can connect to the server at 'irc.perl.org' and talk to this
      person for help: SineSwiper.

 Bugs / Feature Requests

    Please report any bugs or feature requests via
    https://github.com/SineSwiper/Dist-Zilla-TravisCI/issues.

AUTHOR

    Brendan Byrd <bbyrd@cpan.org>

CONTRIBUTORS

      * Alex Peters <lxp@cpan.org>

      * Graham Knop <haarg@haarg.org>

      * Kent Fredric <kentfredric@gmail.com>

      * Peter Roberts <me+dev@peter-r.co.uk>

      * Randy Stauner <randy@magnificent-tears.com>

      * Torsten Raudssus <torsten@raudss.us>

      * Yasutaka ATARASHI <yakex@cpan.org>

COPYRIGHT AND LICENSE

    This software is Copyright (c) 2015 by Brendan Byrd.

    This is free software, licensed under:

      The Artistic License 2.0 (GPL Compatible)



CHANGES
Changelog for Dist-Zilla-TravisCI

1.14 2015-03-20T03:37:57
 - Remove File::Slurp (wasn't even using it now, anyway) (fixes #20)
   (Brendan Byrd)
 - Suppport Travis CI containerisation infrastructure (Peter Roberts)
 - Removed duplicated code generated for .travis.yml (haargs travis perl
   helpers) (ugexe (Nick Logan))

1.13 2015-02-09T05:31:57
 - Fix precedence bug with poll times in TestRelease (Brendan Byrd)
 - Fix failure to function when untracked files are present (Peter Roberts)
 - Remove dependency on Moose::Autobox (Kent Fredric)
 - Load CommandHelper lazily. (Kent Fredric)

1.12 2014-08-02T23:37:29
 - Detect the proper build in a created state using /builds (Brendan Byrd)
 - Turn off AUTHOR_TESTING for build tests (Brendan Byrd)
 - Fix TravisYML to work with latest Perl 5.20 + blead (Brendan Byrd)
 - Move dirty branch pushing code into its own method in ::Git::Bundle
   (Brendan Byrd)
 - Add AUTHORITY comment to modules (Brendan Byrd)
 - Add version dep to List::AllUtils (for 'pairs') (Brendan Byrd)
 - Add open_status_url option to Travis::TestRelease (Brendan Byrd)
 - Add --author flag to `dzil listdeps' in .travis.yml. (Yasutaka ATARASHI)

1.11 2014-08-02T21:57:27
 - Turn off AUTHOR_TESTING for build tests (Brendan Byrd)
 - Fix TravisYML to work with latest Perl 5.20 + blead (Brendan Byrd)
 - Move dirty branch pushing code into its own method in ::Git::Bundle
   (Brendan Byrd)
 - Add AUTHORITY comment to modules (Brendan Byrd)
 - Add version dep to List::AllUtils (for 'pairs') (Brendan Byrd)
 - Add open_status_url option to Travis::TestRelease (Brendan Byrd)

1.10 2014-05-02T04:18:45
 - Fix ChainSmoking.pm (Brendan Byrd)
 - Write post-failure comment for #11 (Brendan Byrd)
 - Many additions/fixes for TravisYML: (Brendan Byrd)
 - New plugin: Travis::TestRelease! (Brendan Byrd)

1.05 2014-04-21T13:24:59
 - Define dump_config for compatibility with newer Git::Repo (Randy
   Stauner)
 - Remove --skip-satisfied in favor of listdeps --missing (#14) (Brendan
   Byrd)
 - fix typo in log message (Alex Peters)
 - Use two-pass install method for MVDT (#9) (Brendan Byrd)

1.04 2013-06-26T00:05:27
 - Remove Perl 5.8 support (for now) (Brendan Byrd)

1.03 2013-06-23T02:35:24
 - Move around Moose/sanity to (try to) prevent experimental warnings
   (Brendan Byrd)
 - Add new Perl versions recently added to Travis CI:	 5.19, 5.18, 5.8
   (Brendan Byrd)

1.02 2013-06-13T13:26:30
 - Remove Dwarn debug from Plugin (no idea how that passed tests...)
   (Brendan Byrd)

1.01 2013-06-08T01:09:56
 - Fix fatal bug with build branch (Brendan Byrd)
 - Fix docs (Brendan Byrd)

0.98 2013-06-07T11:54:32
 - Fix problem with build-branch .travis.yml not getting properly    added
   if only release_branch is turned on in Git::CommitBuild (Brendan Byrd)

0.97 2013-04-23T02:59:59
 - Fix up docs a bit (fixes Issue #3) (Brendan Byrd)
 - Remove weaver.ini (now in PW BBYRD bundle) (Brendan Byrd)
 - Fix email missing "Please tell me who you are" errors (Brendan Byrd)
 - I don't think it should be really doing AUTHOR and RELEASE testing for
   the dependencies (Torsten Raudssus)

0.96 2013-03-19T18:24:59
 - Prevent "Please tell me who you are" errors for certain DZIL configs
   (Brendan Byrd)
 - Fix typo on IRC block in YML (Brendan Byrd)

0.95 2012-10-02T19:46:10
 - Fix typo on IRC block in YML (Brendan Byrd)

0.94 2012-09-28T20:17:42
 - Fix build branch to include 'installdeps' command (Brendan Byrd)

0.93 2012-09-28T19:54:26
 - Chain smoking for local branch master (Brendan Byrd)
 - Bug fixes (Brendan Byrd)
 - Superfast parallelized dependancy installs for "dzil authordeps"
   (Brendan Byrd)
 - Minor fixes (Brendan Byrd)

0.92 2012-09-06T16:20:49
 - Fix issue #2 (repo_root problem) (Brendan Byrd)

0.91 2012-08-01T15:08:51
 - Remove non-module lines from "| cpanm" commands (like    ReportPhases'
   "Current Command") (Brendan Byrd)

0.90 2012-07-28T13:56:43
 - Minor fixes (Brendan Byrd)
 - Initial commit (Brendan Byrd)



Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.031.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Travis CI plugins for your DZIL-based distribution",
  "AUTHOR" => "Brendan Byrd <bbyrd\@cpan.org>",
  "CONFIGURE_REQUIRES" => {},
  "DISTNAME" => "Dist-Zilla-TravisCI",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "MIN_PERL_VERSION" => "5.010001",
  "NAME" => "Dist::Zilla::TravisCI",
  "PREREQ_PM" => {
    "Browser::Open" => "0.01",
    "Date::Format" => 0,
    "Date::Parse" => "2.29",
    "Dist::Zilla" => "1.093220",
    "Dist::Zilla::Role::Git::DirtyFiles" => "1.100740",
    "Dist::Zilla::Role::Git::Remote" => "0.1.1",
    "Dist::Zilla::Role::Git::Remote::Branch" => "0.1.1",
    "Dist::Zilla::Role::Git::Remote::Check" => "0.1.1",
    "Dist::Zilla::Role::Git::Repo" => "1.113220",
    "Dist::Zilla::Role::MetaCPANInterfacer" => "0.90",
    "Dist::Zilla::Util::Git::Wrapper" => "0.001000",
    "File::pushd" => "1.001",
    "List::AllUtils" => "0.01",
    "Moose" => "0.34",
    "MooseX::Has::Sugar" => "0.0100",
    "MooseX::Types::Moose" => 0,
    "Net::Travis::API::UA" => "0.001000",
    "Try::Tiny" => "0.01",
    "YAML" => "0.67",
    "sanity" => "0.91"
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla" => "1.093220",
    "List::AllUtils" => "0.01",
    "Test::CheckDeps" => "0.010",
    "Test::DZil" => "4.200002",
    "Test::More" => "0.94",
    "Test::Most" => "0.01"
  },
  "VERSION" => "1.14",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Browser::Open" => "0.01",
  "Date::Format" => 0,
  "Date::Parse" => "2.29",
  "Dist::Zilla" => "1.093220",
  "Dist::Zilla::Role::Git::DirtyFiles" => "1.100740",
  "Dist::Zilla::Role::Git::Remote" => "0.1.1",
  "Dist::Zilla::Role::Git::Remote::Branch" => "0.1.1",
  "Dist::Zilla::Role::Git::Remote::Check" => "0.1.1",
  "Dist::Zilla::Role::Git::Repo" => "1.113220",
  "Dist::Zilla::Role::MetaCPANInterfacer" => "0.90",
  "Dist::Zilla::Util::Git::Wrapper" => "0.001000",
  "File::pushd" => "1.001",
  "List::AllUtils" => "0.01",
  "Moose" => "0.34",
  "MooseX::Has::Sugar" => "0.0100",
  "MooseX::Types::Moose" => 0,
  "Net::Travis::API::UA" => "0.001000",
  "Test::CheckDeps" => "0.010",
  "Test::DZil" => "4.200002",
  "Test::More" => "0.94",
  "Test::Most" => "0.01",
  "Try::Tiny" => "0.01",
  "YAML" => "0.67",
  "sanity" => "0.91"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



